core-primitives/stf-interface/src/mocks.rs

```diff
@@ -22,9 +22,67 @@ use crate::{
 	system_pallet::SystemPalletAccountInterface, ExecuteCall, ExecuteGetter, InitState,
 	StateCallInterface, StateGetterInterface, UpdateState,
 };
-use alloc::{string::String, vec::Vec};
+use alloc::{boxed::Box, string::String, vec::Vec};
 use core::marker::PhantomData;
-use itp_types::OpaqueCall;
+use itp_types::{AccountId, OpaqueCall};
+
+#[cfg(feature = "std")]
+pub use std_only::*;
+
+#[cfg(feature = "std")]
+mod std_only {
+	use super::*;
+	use lazy_static::lazy_static;
+	use std::sync::RwLock;
+
+	lazy_static! {
+		/// Global counter for event access.
+		pub static ref EVENT_HANDLER: RwLock<EventCounter> = RwLock::new(EventCounter::default());
```

Comment:
Currently, in std only, because I didn't want to introduce an unused sgx feature. Since most crates will be available in std (including executor and such) soon, I don't think that a sgx feature will be needed in the future.

Reply:
What is the idea and reasoning behind the global variable here? Especially in a mock implementation, I would not expect to see any global state (can mess up unit tests really badly). Is there no way of making this local to the instance?

Reply:
And on the topic for `std`/`sgx`, all you need to do, is have two different `use` statements for `RwLock`, right? I think that's well worth the effort (as we have already in many places).

Reply:
Didn't see any other way to test if [Stf::reset_events](https://github.com/integritee-network/worker/blob/bh/get-events/core/offchain-worker-executor/src/executor.rs#L122) is called. Other than introducing a panic. Happy to hear your adive if you have a better idea.

Reply:
Usually that's done by introducing state into the mock. We can discuss further ☺️

Reply:
> And on the topic for std/sgx, all you need to do, is have two different use statements for RwLock, right? I think that's well worth the effort (as we have already in many places).

Except that stf-interface is for now not relying on sgx-features. So I want to avoid introducing an sgx feature just because of a mock, which is not even used by sgx yet. But lets talk tomorrow.

Reply:
@murerfel better now?

Reply:
Yes, looking good now 🎉 👍